rfc/monorepo.md

```diff
@@ -0,0 +1,557 @@
+    - Existing single-project forks can also contribute changes back with minor workflow changes
+    - If an existing fork is itself converted into a monorepo via the subtree method, it can then pull directly from the main project.
+
+
+There are some downsides, though:
+
+1. Existing web links to pulls or issues will be broken.  However, links to indivdual commits will actually still work just fine.
+    - `https://github.com/spinnaker/clouddriver/pull/5856` does NOT translate to `https://github.com/spinnaker/spinnaker/pull/5856`
+    - `https://github.com/spinnaker/clouddriver/commit/46706fec79b0e909b8e99516f3328067f1504ad4` DOES translate to `https://github.com/spinnaker/spinnaker/commit/46706fec79b0e909b8e99516f3328067f1504ad4`
+
+1. Subtree-merges performed to create the monorepo will not have their child commits rendered properly in the Github UI.  All commits are still present and directly addressable (as above), but Github does not follow the subtree pseudo-rename in the UI.  Github will report tens of thousands of commits in the entire tree, but only show the top-level commits for each subtree-imported project - future commits will be fully traversable in the Github UI as expected.
+
+### Subtree Merges and Contributions From Existing Forks
+
+This section details how to integrate changes between a single project repository (e.g. an existing OSS project or a private fork) and a subtree-ified monorepo of all Spinnaker projects.
+
+Heavy use is made of Git's [`subtree` merge strategy](https://git-scm.com/docs/merge-strategies#Documentation/merge-strategies.txt-subtree).
+
+Expects that you have [SSH-based Github authentication](https://docs.github.com/en/authentication/connecting-to-github-with-ssh) configured on your local machine.
+
+#### Creation of the OSS Monorepo
+
+A subtree-ified monorepo of the entire Spinnaker project can be created with the following script, following the `git init` command in your folder of choice:
+
+```sh
+#!/usr/bin/env sh
+
+# Subtree clones
+git subtree add -P clouddriver git@github.com:spinnaker/clouddriver.git master
+git subtree add -P deck git@github.com:spinnaker/deck.git master
+git subtree add -P deck-kayenta git@github.com:spinnaker/deck-kayenta.git master
+git subtree add -P echo git@github.com:spinnaker/echo.git master
+git subtree add -P fiat git@github.com:spinnaker/fiat.git master
+git subtree add -P front50 git@github.com:spinnaker/front50.git master
+git subtree add -P gate git@github.com:spinnaker/gate.git master
+git subtree add -P halyard git@github.com:spinnaker/halyard.git master
+git subtree add -P igor git@github.com:spinnaker/igor.git master
+git subtree add -P kayenta git@github.com:spinnaker/kayenta.git master
+git subtree add -P keel git@github.com:spinnaker/keel.git master
+git subtree add -P kork git@github.com:spinnaker/kork.git master
+git subtree add -P orca git@github.com:spinnaker/orca.git master
+git subtree add -P rosco git@github.com:spinnaker/rosco.git master
```

Comment:
[spinnaker-monitoring](https://github.com/spinnaker/spinnaker-monitoring) is [below](https://github.com/spinnaker/governance/pull/336/files#diff-a56c8a7010c73454be672ac2b81d8df1e145318115a9a833fa9570da4b1dc36cR329), but not here....It's still in [the bom](https://github.com/spinnaker/buildtool/blob/f7c125019e4b7e67e9bf7f182ae4963fc854fdf9/dev/buildtool/bom_scm.py#L36), and not technically archived just yet.  Worth some discussion?

Reply:
Definitely makes sense to include it until it's officially replaced - I'll add it here as well.

Reply:
> I think we should NOT include spinnaker-monitoring It's not even referenced on the docs as a supported system anymore. Just hasn't been removed from halyard... and probably should figure out how to make sure it's defaulted to observability plugin instead. BUT kinda been waiting to see where "how-to-deploy-spinnaker" stuff goes

I do agree that `spinnaker-monitoring` is de-facto replaced for most major users, but we can always delete it from the monorepo once it's actually replaced.  I think it is likely to be more total effort to exclude it, as that would complicate the build and release process until it is removed.

Reply:
Yes, `spinnaker-monitoring` repo is frozen.
The version 1.4.0 is pinned in Halyard's BOM.
During a release we re-tag the existing container image with the friendly `spinnaker-1.x.y` so it's easy to consume.

We can continue to do something similar in monorepo CI until support for it is removed.
We don't need to and shouldn't include it in a monorepo.

Reply:
If it isn't built as part of a release, then I agree it should be excluded.  I'll revise to that effect.